src/nspawn/nspawn.c

```diff
@@ -1070,6 +1071,9 @@ static int parse_argv(int argc, char *argv[]) {
         parse_share_ns_env("SYSTEMD_NSPAWN_SHARE_NS_UTS", CLONE_NEWUTS);
         parse_share_ns_env("SYSTEMD_NSPAWN_SHARE_SYSTEM", CLONE_NEWIPC|CLONE_NEWPID|CLONE_NEWUTS);
 
+        if (getenv_bool("SYSTEMD_NSPAWN_MOUNT_RW") <= 0)
+                mount_settings |= MOUNT_RO;
```

Comment:
I'd probably turn this around, and set mount_settings to MOUNT_RO when declaring the variable (instead of setting it to 0 then), and then remove the bit, if the env var here is set. I think it's a good idea to have the static arg_xyz variables all initialized to their actual defaults by default, in order to make things more obvious to readers.